src/WCSimWCSD.cc

```diff
@@ -163,11 +164,16 @@ G4bool WCSimWCSD::ProcessHits(G4Step* aStep, G4TouchableHistory*)
      G4double local_y = localPosition.y();
      G4double local_z = localPosition.z();
      theta_angle = acos(fabs(local_z)/sqrt(pow(local_x,2)+pow(local_y,2)+pow(local_z,2)))/3.1415926*180.;
-     effectiveAngularEfficiency = fdet->GetPMTCollectionEfficiency(theta_angle, collectionName[0]);
+     effectiveAngularEfficiency = fdet->GetPMTCollectionEfficiency(theta_angle, volumeName);
      if (G4UniformRand() <= effectiveAngularEfficiency || fdet->UsePMT_Coll_Eff()==0){
-
+       //Retrieve the pointer to the appropriate hit collection. Since volumeName is the same as the SD name, this works. 
```

Comment:
Yes, currently collection name, SD name and volume name are all the same. The collection name and volume name are the same so that you can find which volume you are in and easily assign it to the proper collection. Also, when the WCSimGeometryTable code assigns PMT numbers, you want it to do it for both the ID and OD PMT volumes. It is possible, in principle, for volume name and collection name to be different, but you would then need to keep track of both and have the user define both in the DetectorConfigs. For the sensitive detector, I just wanted it to be distinct for the different volumes and making it equal to the hits collection achieves this.